apps/shop/tests.py

```diff
@@ -6,34 +9,168 @@
 from apps.authentication.models import Email, OnlineUser
 from apps.inventory.models import Item
 from apps.oauth2_provider.test import OAuth2TestCase
+from apps.payment.models import PaymentTransaction
+from apps.payment.transaction_constants import TransactionSource
 from apps.shop.models import MagicToken
 
 
-class ShopAPIURLTestCase(APITestCase):
-    def test_item_list_empty(self):
-        url = reverse("item-list")
+class ShopItemTestCase(APITestCase):
+    basename = "shop_inventory"
+
+    def get_list_url(self):
+        return reverse(f"{self.basename}-list")
 
-        response = self.client.get(url)
+    def get_detail_url(self, _id):
+        return reverse(f"{self.basename}-detail", args=(_id,))
+
+    def test_item_list_empty(self):
+        response = self.client.get(self.get_list_url())
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
     def test_item_list_exists(self):
         G(Item, available=True)
-        url = reverse("item-list")
-
-        response = self.client.get(url)
+        response = self.client.get(self.get_list_url())
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
     def test_item_detail(self):
         item = G(Item, available=True)
-        url = reverse("item-detail", args=(item.id,))
 
-        response = self.client.get(url)
+        response = self.client.get(self.get_detail_url(item.id))
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
 
+class ShopOrderLinesTestCase(OAuth2TestCase):
+    scopes = ["shop.readwrite", "read", "write"]
+    basename = "shop_order_lines"
```

Comment:
Shouldn't this be linked to the actual basename, so that if you wanted to update it you would only have to do it one place?

Reply:
The way Django works wont let you do that in all instances anyways, so I don't really see it as that important.